irohad/ametsuchi/impl/storage_impl.hpp

```diff
@@ -119,20 +110,9 @@ namespace iroha {
       ~StorageImpl() override;
 
      protected:
-      struct PoolWrapper {
-        PoolWrapper(std::shared_ptr<soci::connection_pool>,
-                    std::unique_ptr<FailoverCallbackFactory>);
-
-        PoolWrapper(PoolWrapper &&) = default;
-        ~PoolWrapper() = default;
-
-        std::shared_ptr<soci::connection_pool> connection_pool_;
-        std::unique_ptr<FailoverCallbackFactory> failover_callback_factory_;
-      };
-
       StorageImpl(PostgresOptions postgres_options,
                   std::unique_ptr<KeyValueStorage> block_store,
-                  PoolWrapper pool_wrapper,
+                  std::unique_ptr<PoolWrapper> pool_wrapper,
```

Comment:
Not sure if another level of indirection is required, because the struct itself stores two pointers and a bool.